### PR TITLE
Improvement, bonus discovery / depletion xml edits

### PR DESCRIPTION
**Pull request by dzfischer**
WIP; got naval done, added all discovery to settlement line (sans mammoth), fixed some build ordering mistakes.
Ought to finish tomorrow
Edit: lol...

### CODE REVIEW

File: Sources/CvPlot.cpp [hunk 1]
```diff
@@ -710,31 +710,43 @@ void CvPlot::doTurn()
...
+
+	// Bonus density normalization (use land tiles, account for bonus type?)
+	iOdds *= 7 * (GC.getMap().getNumBonuses((BonusTypes) iBonus) + 2);
+	iOdds /= 2 * (GC.getMap().getWorldSize() + 9);
+	
+	return iOdds;
+}
+
+
 void CvPlot::doImprovement()
```
Comment (dzfischer) [inline Sources/CvPlot.cpp:729]: Forgot I had this commit from earlier; seems to work fine, but maybe have a more experienced cpp person glance at it to make sure before v41.1?

**Comment by dzfischer** [comment]
This is ready for squash/merge, though best get Toffer or Matt or someone to take a quick look that I pulled out the resource discovery odds correctly. Alternatively just drop that bit, shouldn't matter for the meat of this pr (xml edits to discovery chances; run the py script to see a breakdown...)

**Comment by raxo2222** [comment]
Is it going anywhere?

**Comment by flabbert** [comment]
@Toffer90 can you have a look at this and say if we want to merge or kill?

**Pull request closed without merging**